utils/monitoring.js

```diff
@@ -40,7 +37,7 @@ const objectsCount = new promClient.Counter({
  */
 function _writeResponse(res, error) {
     let statusCode = 500;
-    if (error instanceof errors.ArsenalError && Number.isInteger(error.code)) {
+    if (error && Number.isInteger(error.code)) {
```

Comment:
why would it crash when error is not of type `ArsenalError` ?
this should just cause the `error instanceof errors.ArsenalError` to be false, and thus keep the  default statusCode 500...

Reply:
Not sure exactly why, but I suspect there is something odd on the platform where we had the issue (not discussing it here - it's a public repo), we got:
```
{"name":"S3Utils::CountItems::Master","time":1717691607622,"error":{"message":"Right-hand side of 'instanceof' is not an object","stack":"TypeError: Right-hand side of 'instanceof' is not an object\n    at _writeResponse (/usr/src/app/utils/monitoring.js:43:15)\n    at Object.metricsHandler (/usr/src/app/utils/monitoring.js:72:16)","name":"TypeError"},"level":"error","message":"Uncaught Exception","hostname":"artesca-data-ops-count-items-111111-phm8q","pid":12}
```

I suspect this is the same issue that require us to put the `useUnifiedTopology` explicitly, even if this should be true by default in the cluster. There is something wrong somewhere in their setup.

Meanwhile, we do not need to ensure it's an arsenal error anyway, so just removing this code, it will unblock the client till we understand the issue better (if we want to)

Reply:
And just to complete,a s said in the ticket, I was not able to reproduce the issue, and customers who did the upgrade are not complaining for now either.

Reply:
looking at the error : `Right-hand side of 'instanceof' is not an object`, it would seem that this is not a platform issue but really a coding bug

i.e. what is really `errors.ArsenalError` ? is it really defined (or a typo) ?
did you try (in some actually executed code path) to execute the same expression (`error instanceof errors.ArsenalError`) ?

Reply:
> This check is not something we want to have
if we don't need to have it, indeed it can and should be removed.

> but we do not need this check in any case, so not worth digging into it
sorry to disagree, but it is worth digging. We have some difference in production, supposedly with the same (immutable!) docker image: so there is really an issue worth investigating....

Reply:
Yes, not saying we shouldn't try to understand why we have that, but it should be investigated globally, not for this particular bugfix

Reply:
I created https://scality.atlassian.net/browse/OS-832 to track this problem.